Sources/NIOCore/NIOOnLoopSendable.swift

```diff
@@ -0,0 +1,78 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+public struct NIOOnLoopSendable<Value>: @unchecked Sendable {
```

Comment:
Can you please add documentation comments to these that explains what they're for, how to use them, and how they differ?

Reply:
Yes, I will once we settle on a name

Reply:
done